script.js

```diff
@@ -35,6 +11,42 @@ jQuery(function () {
         });
     });
 
-    figureManager.update();
 });
 
+// Figures
+jQuery(function () {
+    const $figureContainer = $('.figures');
+    let maxIndex = 0;
+
+    const addRow = (index, path = '', cation = '') => {
+        maxIndex = index;
+        return '<div class="row">' +
+            '<div class="col-6"><input type="text" class="form-control" name="problem[figures][' + index + '][path]" value="' + path + '"/></div>' +
+            '<div class="col-6"><input type="text" class="form-control" name="problem[figures][' + index + '][caption]" value="' + cation + '"/></div>' +
+            '</div>';
+    };
+
+    let html =  $.map( figures, (figure, index) => {
```

Comment:
Preco nevyuzijes Array.prototypa.map? Je rychlejsie a priamo v JS.

Reply:
Nebylo to pole, ale objekt, tak jsem to musel změnit.